src/commands/apextests/list.ts

```diff
@@ -59,22 +57,25 @@ export default class ApextestsList extends SfCommand<ApextestsListResult> {
     const files = readDir.filter((file) => file.endsWith('.cls') || file.endsWith('.trigger'));
     const testMethodsNames: string[] = [];
 
-    // read each file and check for the test methods at the top
-    files.map((file) => {
-      const data = fs.readFileSync(`${directory}/${file}`, 'utf8');
-
-      // try to find, with a RegEx, the test methods listed at the top of the
-      // file with @Tests or @TestSuites
-      const testMethods = data.match(/(@(Tests|TestSuites)).+/g);
-      // for each entry, parse the names
-      // const testMethodsNames = testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : [];
-      testMethodsNames.push(...(testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : []));
+    const readPromises = files.map(async (file) => {
```

Comment:
**suggestion:** instead of iterating over every file and spawn an async job doing IO all in parallel, you could use a parallel working queue and ensure it does not stress the system more than 6 by 6 per exemple
It could be an issue to not control the concurrency for very large repository

Exemple of implementation [here](https://github.com/scolladon/sfdx-git-delta/blob/main/src/service/diffLineInterpreter.ts)